### DISCUSSION TITLE
CVE-2024-28085 fix for older versions of util-linux

### DISCUSSION THREAD

**Discussion by sshedi** [util-linux/util-linux#2868] [General]
https://github.com/util-linux/util-linux/commit/404b0781f52f7c045ca811b2dceec526408ac253

Hi @karelzak Are there any plans to patch CVE-2024-28085 in older releases of util-linux? Many distros will not be using the latest version of it and `fputs_careful` got introduced in a fairly newer version of util-linux.

Is it okay for you to provide a fix for older versions of util-linux?

**Comment by karelzak** [comment]
I plan a 2.39.4 release. For the older versions, we can add a patch to the old branches; I don't think releases are necessary.

**Comment by sshedi** [comment]
That would be really nice. Thanks a lot for the response.

Will be waiting for the patches, any info on the timeline of the fix?

**Comment by karelzak** [comment]
The patch has been merged into stable/v2.39.1 (https://github.com/util-linux/util-linux/commit/a98558aef1e347adba2cb0028c036ce156aa68a7).  I want to release it on Wednesday.

**Comment by sshedi** [comment]
Thanks @karelzak for the fix. I was hoping you would provide a fix for 2.39

https://github.com/util-linux/util-linux/blob/v2.39/term-utils/wall.c#L331

I think 2.39 and 2.39.1 have a big difference in code with print modules.

**Comment by karelzak** [comment]
The wall.c is almost the same in stable/v2.39 and stable/v2.40 (except for the systemd support).
https://github.com/util-linux/util-linux/blob/stable/v2.39/term-utils/wall.c#L331

**Comment by sshedi** [comment]
Please extract and check the tarball
http://archive.ubuntu.com/ubuntu/pool/main/u/util-linux/util-linux_2.34-0.1ubuntu9.5.debian.tar.xz

Ubuntu added 3 prep patches to fix this CVE.

I didn't realise that v2.39 tag and stable/v2.39 are fully different.

**Comment by karelzak** [comment]
> Please extract and check the tarball http://archive.ubuntu.com/ubuntu/pool/main/u/util-linux/util-linux_2.34-0.1ubuntu9.5.debian.tar.xz

This is v2.34; I'm talking about v2.39, where all the `include/carefulputc.h` patches should be already included (since v2.39-rc2).
